src/db/jobs.rs

```diff
@@ -0,0 +1,104 @@
+//! The `jobs` table provides a way to have scheduled jobs
+use anyhow::{Result, Context as _};
+use chrono::{DateTime, FixedOffset};
+use tokio_postgres::{Client as DbClient};
+use uuid::Uuid;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Job {
+    pub id: Uuid,
+    pub name: String,
+    pub expected_time: DateTime<FixedOffset>,
+    pub metadata: serde_json::Value,
+    pub executed_at: Option<DateTime<FixedOffset>>,
+    pub error_message: Option<String>,
+}
+
+pub async fn insert_job(db: &DbClient, job: &Job) -> Result<()> {
+    tracing::trace!("insert_job(id={})", job.id);
+    
+    db.execute(
+        "INSERT INTO jobs (id, name, expected_time, metadata, executed_at, error_message) VALUES ($1, $2, $3, $4, $5, $6) ON CONFLICT DO NOTHING",
```

Comment:
I would expect something closer to `ON CONFLICT do update set metadata = EXCLUDED.metadata`, so that we can naturally update with a new set of metadata.

I think we probably ought to change `Job` to not have an ID field, and instead have `id` returned from this function -- that way, when you queue up work to happen at X time, you naturally get an ID back (whether a new one or an existing one).

Reply:
That definitely sounds better 👍🏼